template/query.py

```diff
@@ -49,6 +72,20 @@ def select(self, key, column, query_columns):
     # Returns False if no records exist with given key or if the target record cannot be accessed due to 2PL locking
     """
     def update(self, key, *columns):
+        # take a snapshot?
+        # all updates will go to the tail page
+        # 1. Check that key exists in page directory else return false
+        # 2. Create a new record, with a new rid
+        # 3. We will be given a list [None, None, NewValue, ... None, None]
+        # 4. Update schema encoding at all columns with non None values to equal 1
+        # 5. New record will go into our tail page
+        # 6. Update the page_dictory with the rid of the MRU
+        # TODO : where do tail records go in our table?
```

Comment:
Page directory will just hold the base RID of the record, the indirection column of that record will tell us where to go to find the tail page record. that will store the relative tail page and column index of the tail page.=, which is all we need once we are at the base page level.